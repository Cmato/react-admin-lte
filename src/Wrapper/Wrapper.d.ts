import * as React from 'react';

export interface WrapperProps {}

declare const Wrapper: React.ComponentType<WrapperProps>;

export default Wrapper;
