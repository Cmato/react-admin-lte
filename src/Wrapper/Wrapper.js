import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Wrapper = ({
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    wrapper: true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

Wrapper.propTypes = {
  /**
   * The content of the wrapper.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

export default Wrapper;
