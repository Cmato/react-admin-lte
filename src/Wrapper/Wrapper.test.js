import React from 'react';
import { shallow } from 'enzyme';
import Wrapper from './Wrapper';

it('renders correctly', () => {
  expect(shallow(<Wrapper className="best-wrapper"><div /></Wrapper>))
    .toMatchSnapshot();
});
