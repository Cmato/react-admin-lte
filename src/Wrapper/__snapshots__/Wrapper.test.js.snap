// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper best-wrapper"
>
  <div />
</div>
`;
